Fix eviction policy enum values to use hyphens instead of underscores

jkpe opened the pull request:
Fixes the `eviction_policy_model` in the API spec.

Example of it failing with the current values:

```
curl -X PATCH \
  -H "Content-Type: application/json" \
  -H "Authorization: Bearer dop_x" \
  -d '{
  "config": {
    "redis_maxmemory_policy": "volatile_lru"
  }
}' \
  "https://api.digitalocean.com/v2/databases/f1900b2d-688b-4898-9eef-83c48df855b2/config"
{"message":"invalid eviction policy - volatile_lru","id":"bad_request","request_id":"d7c7156a-e850-49ac-abed-fb41d8bf0367"}
```

andrewsomething reviewed (changes requested):
Unfortunately, this is an inconsistency in the API. 

You are updating the docs for the `/v2/databases/$DATABASE_ID/eviction_policy`, but your example of a failed request is for the `/v2/databases/$DATABASE_ID/config` endpoint. These strings take different formats for the two endpoints. For example:

`eviction_policy` fails with `volatile-lru` but succeeds with `volatile_lru`
```
asb@asb-do-laptop:~$ curl -X PUT \
  -H "Content-Type: application/json" \
  -H "Authorization: Bearer $DO_TOKEN" \
  -d '{"eviction_policy":"volatile_lru"}' \
  "https://api.digitalocean.com/v2/databases/d179faff-2f98-4858-aa45-8bac959f4464/eviction_policy"
asb@asb-do-laptop:~$ curl -X PUT \
  -H "Content-Type: application/json" \
  -H "Authorization: Bearer $DO_TOKEN" \
  -d '{"eviction_policy":"volatile-lru"}' \
  "https://api.digitalocean.com/v2/databases/d179faff-2f98-4858-aa45-8bac959f4464/eviction_policy"
{"message":"invalid eviction policy","id":"unprocessable_entity","request_id":"3ff6340b-d4f5-4684-bdc2-80fb4cc0a9be"}
```
`config` fails with `volatile_lru` but succeeds with `volatile-lru`

```
asb@asb-do-laptop:~$ curl -X PATCH \
  -H "Content-Type: application/json" \
  -H "Authorization: Bearer $DO_TOKEN" \
  -d '{"config": {"redis_maxmemory_policy": "volatile_lru"}}' \
  "https://api.digitalocean.com/v2/databases/d179faff-2f98-4858-aa45-8bac959f4464/config"
{"message":"invalid eviction policy - volatile_lru","id":"bad_request","request_id":"f0572386-a37a-488b-be51-123c76ee066c"}
asb@asb-do-laptop:~$ curl -X PATCH \
  -H "Content-Type: application/json" \
  -H "Authorization: Bearer $DO_TOKEN" \
  -d '{"config": {"redis_maxmemory_policy": "volatile-lru"}}' \
  "https://api.digitalocean.com/v2/databases/d179faff-2f98-4858-aa45-8bac959f4464/config"
```

Ideally the DBaaS API would normalize these values and accept either for both endpoints, but unfortunately that is not the case today.

jkpe commented:
> Unfortunately, this is an inconsistency in the API.

Ahh, that is a tad confusing. Thanks @andrewsomething 

I've gone ahead and reverted my original change and replaced the reference to `eviction_policy_model.yml` with corrected values for that endpoint.

Let me know if you'd prefer a new PR, or to replace my commit history so it is cleaner? Or if you're going to squash and merge I suppose it should be OK.

andrewsomething reviewed (approved):
Great catch! I didn't realize we were sharing the model between the two endpoint. 

> Let me know if you'd prefer a new PR, or to replace my commit history so it is cleaner? Or if you're going to squash and merge I suppose it should be OK.

By default, we squash commits. So this is fine.

Thanks for improving the API docs! :pray:
